### DISCUSSION TITLE
Is there any document to describe the threads in a Participant?

### DISCUSSION THREAD

**Discussion by wjhtinger** [eProsima/Fast-DDS#3686] [General]
I run the DDSHelloWorldExample, found the below threads are created. So, who can tell the roles and functions of these threads ?
Appreciate.

bash(2551)---DDSHelloWorldEx(4989)-+-{DDSHelloWorldEx}(4990)
           |               |                       |                                    |-{DDSHelloWorldEx}(4991)
           |               |                       |                                    |-{DDSHelloWorldEx}(4992)
           |               |                       |                                    |-{DDSHelloWorldEx}(4993)
           |               |                       |                                    |-{DDSHelloWorldEx}(4994)
           |               |                       |                                    |-{DDSHelloWorldEx}(4995)
           |               |                       |                                    |-{DDSHelloWorldEx}(4996)
           |               |                       |                                    |-{DDSHelloWorldEx}(4997)
           |               |                       |                                    `-{DDSHelloWorldEx}(4998)

**Comment by JesusPoderoso** [comment]
Hi @wjhtinger,
there is a descriptive table in the Fast DDS documentation regarding [concurrency and multithreading](https://fast-dds.docs.eprosima.com/en/latest/fastdds/library_overview/library_overview.html#concurrency-and-multithreading). 
I think that is what you were looking for.

**Reply by wjhtinger** [reply]
Thanks。
A suggestion, just give every thread a different name, then user can check waht threads are created.
